server/src/main/java/org/apache/lucene/codecs/lucene90/StarTree99DocValuesProducer.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.apache.lucene.codecs.lucene90;
+
+import org.apache.lucene.codecs.DocValuesProducer;
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.FieldInfos;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.NumericDocValues;
+import org.apache.lucene.index.SegmentReadState;
+import org.apache.lucene.index.SortedDocValues;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.index.SortedSetDocValues;
+import org.apache.lucene.index.VectorEncoding;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.MetricEntry;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import static org.opensearch.index.compositeindex.datacube.startree.utils.StarTreeHelper.fullFieldNameForStarTreeDimensionsDocValues;
+import static org.opensearch.index.compositeindex.datacube.startree.utils.StarTreeHelper.fullFieldNameForStarTreeMetricsDocValues;
+
+/**
+ * This class is a custom abstraction of the {@link DocValuesProducer} for the Star Tree index structure.
+ * It is responsible for providing access to various types of document values (numeric, binary, sorted, sorted numeric,
+ * and sorted set) for fields in the Star Tree index.
+ *
+ * @opensearch.experimental
+ */
+public class StarTree99DocValuesProducer extends DocValuesProducer {
```

Comment:
nit : Composite99DocValuesProducer ?

Reply:
I feel this is specific to star tree since it inputs dimensions and metric entries

Reply:
so composite99DocValuesConsumer will write but StarTree99DocValuesProducer will read ?

why can't we pass set of unique field names , why do we need to know dimensions and metrics?

Reply:
We can initialize fieldInfos as part of segmentReadState and make this whole class very generic which can be reused for other use cases as well. ( similar to consumer comment, we can make it a generic wrapper class )